### PR TITLE
chore(TSLint): TS 4.0.2 and Eslint 6.5.1

### PR DESCRIPTION
**Pull request by codecaaron**
## Overview
Bump it.

- Monolith - https://github.com/codecademy-engineering/Codecademy/pull/19373/files
- Static Sites - https://github.com/codecademy-engineering/static-sites/pull/313

### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -66,6 +66,14 @@ module.exports = {
     ],
 
     // These rules could be useful, but we haven't gotten around to trying them out
+    // >3.x.x
```
Comment (JoshuaKGoldberg) [inline packages/eslint-config/index.js:69]: Same nit suggestion 😄 
```suggestion
```

File: packages/eslint-config/index.js [hunk 2]
```diff
@@ -66,6 +66,14 @@ module.exports = {
...
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/explicit-module-boundary-types': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    // <3.x.x
```
Comment (JoshuaKGoldberg) [inline packages/eslint-config/index.js:76]:
```suggestion
```

**Review by JoshuaKGoldberg — APPROVED** [review]
You'll probably want to verify this works for client-modules, author, and static-sites as well as the monolith...

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.0.1-alpha.a01b2a.0<br/>@codecademy/gamut-labs@4.1.7-alpha.a01b2a.0<br/>@codecademy/gamut@16.0.10-alpha.a01b2a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f6cbc7df42fd80b04caaf7f--gamut-preview.netlify.app>https://5f6cbc7df42fd80b04caaf7f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f6cbc7df42fd80b04caaf7f>Deploy Logs</a></p>

File: package.json [hunk 3]
```diff
@@ -87,7 +87,7 @@
     "react-docgen-typescript-loader": "3.6.0",
     "react-dom": "16.12.0",
     "react-test-renderer": "16.12.0",
-    "typescript": "3.8.3"
+    "typescript": "4.0.2"
```
Comment (hmafzal) [inline package.json:90]: +1 @JoshuaKGoldberg comment - are there matching PRs consuming this change in the downstream repos? Can we consume it without any new eslint complaints?

**Review by hmafzal — APPROVED** [review]
LGTM

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.0.1-alpha.925287.0<br/>@codecademy/gamut-labs@4.2.1-alpha.925287.0<br/>@codecademy/gamut@16.0.11-alpha.925287.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f6e152dc01294345dbcc72f--gamut-preview.netlify.app>https://5f6e152dc01294345dbcc72f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f6e152dc01294345dbcc72f>Deploy Logs</a></p>

**Comment by JoshuaKGoldberg** [comment]
> TSLint

_eye twitches_

**Pull request merged**